### ISSUE TITLE
Pass "-1" in the url as an int-type parameter

### ISSUE THREAD

**Issue by niti-netro42** [cakephp/cakephp#16640]
### Steps to reproduce

1. Create a function that accept int as a parameter.

```php
public function view(int $id) {
...
}
```

2. Pass `-1` in the url e.g., `http://localhost/blogs/view/-1`

3. This is an error on screen
```
Unable to coerce "-1" to `int` for `id` in action BlogsController::view().
```

### Proposed fix

Changing this line `return ctype_digit($argument) ? (int)$argument : null;` from [ControllerFactory.php#L271](https://github.com/cakephp/cakephp/blob/4.x/src/Controller/ControllerFactory.php#L271)

To:

```php
return filter_var($argument, FILTER_VALIDATE_INT) !== false ? (int)$argument : null;
```

Tested: https://onlinephp.io/c/16865

### CakePHP Version

4.4.1

### PHP Version

7.4

**Comment by othercorey** [comment]
Did we decide that integer parameters were expected to be IDs?

**Comment by othercorey** [comment]
> Did we decide that integer parameters were expected to be IDs?

Looks like we decided against is_numeric() checks for int: https://github.com/cakephp/cakephp/pull/15984#discussion_r719482591

**Comment by markstory** [comment]
> Did we decide that integer parameters were expected to be IDs?

I think that is the general use case, but you can use `-1` as a primary key value in most databases.

**Comment by niti-netro42** [comment]
Hi @markstory , 

I'm not sure that I'm allowed to re-open this issue. I just encountered a similar issue when I pass 0 instead and get same error.

```
Unable to coerce "0" to `int` for `id` in action Blogs::view().
```

Please let me know if I need to create a new issue.

Thanks,

**Comment by ADmad** [comment]
@niti-netro42 "0" coercion issue has already been fixed in [4.4.4](https://github.com/cakephp/cakephp/releases/tag/4.4.4) release.
